tests/e2e/e2e_test.go

```diff
@@ -1,70 +1,71 @@
 package e2e
 
 import (
-	"errors"
 	"fmt"
 	"io"
 	"net/http"
+	"strings"
 	"time"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 )
 
-func (s *IntegrationTestSuite) TestQueryBalances() {
-	var (
-		expectedDenoms    = []string{osmoDenom, stakeDenom}
-		expectedBalancesA = []uint64{osmoBalanceA, stakeBalanceA - stakeAmountA}
-		expectedBalancesB = []uint64{osmoBalanceB, stakeBalanceB - stakeAmountB}
-	)
-
-	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
-	balancesA, err := queryBalances(chainAAPIEndpoint, s.chainA.validators[0].keyInfo.GetAddress().String())
-	s.Require().NoError(err)
-	s.Require().NotNil(balancesA)
-	s.Require().Equal(2, len(balancesA))
-
-	chainBAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainB.id][0].GetHostPort("1317/tcp"))
-	balancesB, err := queryBalances(chainBAPIEndpoint, s.chainB.validators[0].keyInfo.GetAddress().String())
-	s.Require().NoError(err)
-	s.Require().NotNil(balancesB)
-	s.Require().Equal(2, len(balancesB))
-
-	actualDenomsA := make([]string, 0, 2)
-	actualBalancesA := make([]uint64, 0, 2)
-	actualDenomsB := make([]string, 0, 2)
-	actualBalancesB := make([]uint64, 0, 2)
-
-	for _, balanceA := range balancesA {
-		actualDenomsA = append(actualDenomsA, balanceA.GetDenom())
-		actualBalancesA = append(actualBalancesA, balanceA.Amount.Uint64())
-	}
-
-	for _, balanceB := range balancesB {
-		actualDenomsB = append(actualDenomsB, balanceB.GetDenom())
-		actualBalancesB = append(actualBalancesB, balanceB.Amount.Uint64())
-	}
-
-	s.Require().ElementsMatch(expectedDenoms, actualDenomsA)
-	s.Require().ElementsMatch(expectedBalancesA, actualBalancesA)
-	s.Require().ElementsMatch(expectedDenoms, actualDenomsB)
-	s.Require().ElementsMatch(expectedBalancesB, actualBalancesB)
-}
+// func (s *IntegrationTestSuite) TestQueryBalances() {
```

Comment:
I'd prefer if we make this work right away instead of committing dead code that previously worked.

I think for now we should figure out how we can enforce a specific order and let tests run after the other so that we can expect specific values. Later, we can refactor in terms of setup/teardown before each test.

What do you think?

Reply:
Sounds good! I got it to work, but I decided to remove gas fees in order to make calculations simpler (still kept the const so we can add fees if we want in the future)

Reply:
Looking good. Yeah let's add this later

Reply:
Yeah, I'm a bit torn. Typically, the purist in me would state that test execution order should be independent. However, there are exceptions, and I feel like integration/e2e tests where there is a single global state is certainly one of them.

The only way I know for explicitly defining test execution order is by the test name as testify's suite package executes them based on alphabetical order. We can probably achieve this via a prefix and/or suffix to each test name.